### DISCUSSION TITLE
In the event of a kubernetes node failure..

### DISCUSSION THREAD

**Discussion by bshepyard** [strimzi/strimzi-kafka-operator#10061] [Q&A]
In the event of a kubernetes node failure how is it possible for a strimzi operator to recover a kafka broker or zookeeper Pod that was running on that node. Currently the pods just go into terminating state and the operator fails to reconcile the failed broker and zookeeper pods by timing out. This is a bare metal kubernetes setup and I'm using ephemeral storage for the strimzi cluster.

**Comment by scholzj** [comment]
It depends on the exact state of the node and the pod. You can always force delete the pod to continue.

PS: Ephemeral storage is not suitable for any kind of prodution-grade use-cases etc.

**Reply by bshepyard** [reply]
If the node has a hardware failure and is physically powered off is there no automatic way for a kafka broker to be moved to another node without human intervention?

**Reply by scholzj** [reply]
As I said, that cannot be answered easily as it depends on many aspects ... how exactl did the node failed, what storage you use etc. If you use ephemeral storage, it might be simply scheduled on a different node (although you might have a data loss). but it will depend on what Kubernetes does with the Pod etc.

**Reply by bshepyard** [reply]
I am testing the failure of a node so I am powering it off. After a while all deployments and stateful sets restart new pods and the old ones are set to Terminating except the sonar simulator operator which fails to restart as the liveliness and readiness probes fail. And any Kafka broker or zookeeper pod just sits in a Terminating state indefinitely with no new pods created.
[strimzi-pods.txt](https://github.com/strimzi/strimzi-kafka-operator/files/15433983/strimzi-pods.txt)

**Reply by scholzj** [reply]
What Strimzi version are you using? What is the full YAML of the Pod?

**Reply by bshepyard** [reply]
I installed the operator with helm strimzi-kafka-operator-0.40.0
Then I downloaded from the Strimzi site a yaml file to create the cluster see attached.
[kafka-cluster.txt](https://github.com/strimzi/strimzi-kafka-operator/files/15434331/kafka-cluster.txt)

**Reply by scholzj** [reply]
But that is not Pod YAML, so it does not explain the state of the Pod. Strimzi will delete the pod once it terminates -> when it is in failed or succeeded state. There is not much to know about its actual state when it is not terminated. If we delete it, we might cause data loss or other things.

**Reply by bshepyard** [reply]
It wont terminate the Pod until the node comes back online or its manually force deleted!
